storage/backup/backup-cli/src/coordinators/backup.rs

```diff
@@ -358,6 +358,9 @@ impl BackupCompactor {
 
     pub async fn run(&self) -> Result<()> {
         info!("Backup compaction started");
+        // Get a snapshot of remote metadata files before compaction
+        let files = self.storage.list_metadata_files().await?;
+
         // sync the metadata from backup storage
         let mut metaview = metadata::cache::sync_and_load(
```

Comment:
I'd rather return the file list from here -- no chance of inconsistency then.

Reply:
updated to return the file list from metaview